File: 392/solution.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i, j, l1, l2 = 0, 0, len(s), len(t)
        while i < l1 and j < l2:
            if s[i] == t[j]:
                i += 1
            j += 1
        return i == l1

File: 392/Solution.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int i = 0, j = 0, l1 = s.length(), l2 = t.length();
        while (i < l1 && j < l2) {
            if (s.charAt(i) == t.charAt(j)) {
                i++;
            }
            j++;
        }
        return i == l1;
    }
}